### PR TITLE
node: ci: Enable nodefeature test execution in CI for release branches

### PR DESCRIPTION
**Pull request by swatisehgal**
We are adding additional 1,25, 1.26 and 1.27 jobs to ensure that we are not causing regressions by backporting https://github.com/kubernetes/kubernetes/pull/116376 to maintenance releases and to comprehensively test the following backport PRs in CI before the changes are merged:

1. https://github.com/kubernetes/kubernetes/pull/117719
2. https://github.com/kubernetes/kubernetes/pull/116337
3. https://github.com/kubernetes/kubernetes/pull/117738

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
/assign @mmiranda96 

Also please move to sig-node folder

**Comment by swatisehgal** [comment]
@mmiranda96 PR is ready for review.

File: config/jobs/kubernetes/sig-node/k8s-release-branches.yaml [hunk 1]
```diff
@@ -101,3 +101,269 @@ periodics:
...
+        - --repo=github.com/containerd/containerd=main
+        - --service-account=/etc/service-account/service-account.json
+        - --upload=gs://kubernetes-jenkins/pr-logs
+        - --timeout=260
+        - --scenario=kubernetes_e2e
+        - --
+        - --node-args=--image-config-file=/workspace/test-infra/jobs/e2e_node/containerd/containerd-main/cgroupv1/image-config-cgroupv1.yaml
+        - --deployment=node
+        - --gcp-zone=us-west1-b
+        - '--node-test-args=--container-runtime-endpoint=unix:///run/containerd/containerd.sock --container-runtime-process-name=containerd --container-runtime-pid-file= --kubelet-flags="--cgroup-driver=systemd --cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/containerd.service" --extra-log="{\"name\": \"containerd.log\", \"journalctl\": [\"-u\", \"containerd*\"]}"'
```
Comment (mmiranda96) [inline config/jobs/kubernetes/sig-node/k8s-release-branches.yaml:180]: I believe cgroupsv1 doesn't need flag `--cgroup-driver=systemd`.

File: config/jobs/kubernetes/sig-node/k8s-release-branches.yaml [hunk 2]
```diff
@@ -101,3 +101,269 @@ periodics:
...
```
Comment (mmiranda96) [inline config/jobs/kubernetes/sig-node/k8s-release-branches.yaml:268]: Same here.

File: config/jobs/kubernetes/sig-node/k8s-release-branches.yaml [hunk 3]
```diff
@@ -101,3 +101,269 @@ periodics:
...
```
Comment (mmiranda96) [inline config/jobs/kubernetes/sig-node/k8s-release-branches.yaml:356]: Same here.

**Review by mmiranda96 — COMMENTED** [review]
Just a couple of small comments.

**Review by mmiranda96 — COMMENTED** [review]
Just a couple of small comments.

File: config/jobs/kubernetes/sig-node/k8s-release-branches.yaml (hunk 1 above)
Comment (swatisehgal) [reply config/jobs/kubernetes/sig-node/k8s-release-branches.yaml]: Thanks for catching this, addressed.

**Comment by xmudrii** [comment]
@swatisehgal Are we going to need those jobs for 1.28 and onwards? If this is only needed for 1.25-1.27, this is fine. Otherwise, I recommend taking a different approach, i.e. using fork-per-release mechanism (I can further elaborate if that's the way we want to go).

**Comment by swatisehgal** [comment]
> @swatisehgal Are we going to need those jobs for 1.28 and onwards? If this is only needed for 1.25-1.27, this is fine. Otherwise, I recommend taking a different approach, i.e. using fork-per-release mechanism (I can further elaborate if that's the way we want to go).

My intention here was to add these jobs to test backports in CI  like the ones I had (linked in the PR above) and any future backports that get created to release branches under maintenance (which is currently 1.25-1.27) but it would be helpful to have these jobs for 1.28 and onwards when they are under maintenance. 

Please elaborate on the fork per release mechanism as IIUC what you are proposing might be more future proof.

**Comment by xmudrii** [comment]
Here's an example of such a job: https://github.com/kubernetes/test-infra/blob/769121f35fe24711f61ef5a11299bb45b8888939/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L13-L62

You should change `config/jobs/kubernetes/sig-node/k8s-release-branches.yaml` to:

- have only `pull-kubernetes-cos-cgroupv2-containerd-node-e2e-serial` and `pull-kubernetes-cos-cgroupv1-containerd-node-e2e-serial` jobs
  - those jobs should run against master
  - to accomplish that, you need to have `skip_branches` as in the example
  - you need to add `fork-per-release: "true"` annoation

When we add jobs for 1.28, those two jobs will be added as well. However, you'll need to manually add jobs for 1.27-1.25. We add jobs for release branches in files `config/jobs/kubernetes/sig-release/release-branch-jobs/1.2x.yaml` (e.g. `config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml`). You already had that in https://github.com/kubernetes/test-infra/compare/987d4bdb7f71273c443adc6383869a5746b72a89..d61b7dd2e7620a275b72ceff0c3072a35526fab2, so you just need to revert that change.

**Comment by swatisehgal** [comment]
> You should change `config/jobs/kubernetes/sig-node/k8s-release-branches.yaml` to:
> 
> * have only `pull-kubernetes-cos-cgroupv2-containerd-node-e2e-serial and `pull-kubernetes-cos-cgroupv1-containerd-node-e2e-serial` jobs
>   
>   * those jobs should run against master
>   * to accomplish that, you need to have `skip_branches` as in the example
>   * you need to add `fork-per-release: "true"` annoation

Thanks @xmudrii This is very useful information!
Currently, we already have presubmit jobs on the master [pull-kubernetes-cos-cgroupv2-containerd-node-e2e-serial](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml#L1326) and [pull-kubernetes-cos-cgroupv1-containerd-node-e2e-serial](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml#L1414) and I see they have `skip_branches` as you suggested. The only missing piece is the annotation `fork-per-release: "true"`. I will update that and we should be good!

> When we add jobs for 1.28, those two jobs will be added as well. However, you'll need to manually add jobs for 1.27-1.25. We add jobs for release branches in files `config/jobs/kubernetes/sig-release/release-branch-jobs/1.2x.yaml` (e.g. `config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml`). You already had that in https://github.com/kubernetes/test-infra/compare/987d4bdb7f71273c443adc6383869a5746b72a89..d61b7dd2e7620a275b72ceff0c3072a35526fab2, so you just need to revert that change.

Sure, that is not a problem. I moved the jobs because we had a discussion in the SIG Node CI group where it was recommended that since they are SIG-Node related job, I should move them to ` config/jobs/kubernetes/sig-node/k8s-release-branches.yaml` where we already have [sig-node jobs](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-node/k8s-release-branches.yaml) for release branches.

 Are you suggesting that we should move all the jobs (including the currently exiting ones) to  `config/jobs/kubernetes/sig-release/release-branch-jobs/1.2x.yaml` ? 

(cc @SergeyKanzhelev @mmiranda96 to keep them in the loop )

**Comment by xmudrii** [comment]
> Are you suggesting that we should move all the jobs (including the currently exiting ones) to  `config/jobs/kubernetes/sig-release/release-branch-jobs/1.2x.yaml`?

@swatisehgal With `fork-per-release`, jobs will automatically go to `config/jobs/kubernetes/sig-release/release-branch-jobs/1.2x.yaml`, we don't have a way to override the location and say go to `sig-node` folder. It's a drawback, but on the other side, jobs are automatically generated for each release branch. 

> I moved the jobs because we had a discussion in the SIG Node CI group where it was recommended that since they are SIG-Node related job, I should move them to  config/jobs/kubernetes/sig-node/k8s-release-branches.yaml where we already have [sig-node jobs](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-node/k8s-release-branches.yaml) for release branches.

If you want to go with fork-per-release for those jobs as well, we should move those jobs to appropriate release-branch-jobs file as well IMO.

**Comment by swatisehgal** [comment]
> > I moved the jobs because we had a discussion in the SIG Node CI group where it was recommended that since they are SIG-Node related job, I should move them to  config/jobs/kubernetes/sig-node/k8s-release-branches.yaml where we already have [sig-node jobs](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-node/k8s-release-branches.yaml) for release branches.
> 
> If you want to go with fork-per-release for those jobs as well, we should move those jobs to appropriate release-branch-jobs file as well IMO.
 
Ack, Thanks. I think it is more sustainable to have them as fork-per-release for the existing jobs as well. We already need to add there jobs for 1.27. Having said that, I will move my proposed changes under `config/jobs/kubernetes/sig-release/release-branch-jobs/1.2x.yaml` as per your suggestion and we can defer the moving of the existing jobs to be handled in a separate PR.

**Comment by ffromani** [comment]
thanks @swatisehgal ! non-binding LGTM because my limited knowledge of prow config

**Comment by bart0sh** [comment]
@swatisehgal Can you update this PR according to currently supported releases?

**Comment by SergeyKanzhelev** [comment]
This CL is interesting. But needs to be updated to resolve merge conflicts

**Pull request closed without merging**